### ISSUE TITLE
scan-build reports som bugs

### ISSUE THREAD

**Issue by sijohans** [assaabloy-ppi/binson-c-light#23]
I tested using scan-build in my project and got som errors from binson_light.c:

```
$ scan-build gcc -c binson_light.c 
scan-build: Using '/usr/lib/llvm-3.8/bin/clang' for static analysis
binson_light.c:195:3: warning: Undefined or garbage value returned to caller
  return b;
  ^~~~~~~~
binson_light.c:227:42: warning: Access to field 'int_val' results in a dereference of a null pointer (loaded from variable 'val')
      isize = _binson_util_pack_integer( val->int_val, &(pack_buf[1]), (token_type == BINSON_ID_DOUBLE)? 1:0 );
                                         ^~~~~~~~~~~~
binson_light.c:238:36: warning: Dereference of null pointer
      binson_tok_size   tok_size = val->bbuf_val.bsize;
                                   ^~~~~~~~~~~~~~~~~~~
binson_light.c:245:7: warning: Value stored to 'isize' is never read
      isize += tok_size;
      ^        ~~~~~~~~
binson_light.c:259:52: warning: Access to field 'bool_val' results in a dereference of a null pointer (loaded from variable 'val')
      res = _binson_io_write_byte( &(pwriter->io), val->bool_val? BINSON_ID_TRUE : BINSON_ID_FALSE );
                                                   ^~~~~~~~~~~~~
5 warnings generated.
scan-build: 5 bugs found.
scan-build: Run 'scan-view /tmp/scan-build-2018-01-30-075409-17446-1' to examine bug reports.
```

**Comment by ppmag** [comment]
Interesting.... thank you for reporting!

**Comment by ppmag** [comment]
Fixed all above in commit 983fb84fa9dcf5948df8e90a5c13496e2d75364b
